OpenSim/Simulation/SimbodyEngine/BallJoint.h

```diff
@@ -52,14 +52,36 @@ class OSIMSIMULATION_API BallJoint : public Joint {
     OpenSim_DECLARE_CONCRETE_OBJECT(BallJoint, Joint);
 
     /** Specify the Coordinates of the BallJoint */
-    CoordinateIndex rx{ constructCoordinate(Coordinate::MotionType::Rotational) };
-    CoordinateIndex ry{ constructCoordinate(Coordinate::MotionType::Rotational) };
-    CoordinateIndex rz{ constructCoordinate(Coordinate::MotionType::Rotational) };
+    CoordinateIndex rx{constructCoordinate(Coordinate::MotionType::Rotational)};
+    CoordinateIndex ry{constructCoordinate(Coordinate::MotionType::Rotational)};
+    CoordinateIndex rz{constructCoordinate(Coordinate::MotionType::Rotational)};
 
 public:
     /** Use Joint's constructors. @see Joint */
     using Joint::Joint;
 
+    using Joint::getCoordinate;
+
+    /** Indices of Coordinates; can be used with getCoordinate() for
+        convenience. */
+    enum class Coord {
+        Rotation1,
+        Rotation2,
+        Rotation3
```

Comment:
> The names here are not consistent with the names of the `CoordinateIndex`es (`rx`, `ry`, `rz`).

That is true. IMHO, `rz` is not sufficiently descriptive to help the user avoid errors since it is not immediately apparent about _which_ Z-axis the rotation was performed (it could have occurred anywhere in a rotation sequence). `Rotation3`, on the other hand, implies nothing more than what is immediately apparent from the name. Thus, we would be enforcing that the user knows the rotation sequence.

> Is there a benefit to hard-coding these enum values? Is that possible with enum classes?

@klshrinidhi: Can you answer this?